src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,94 +19,224 @@
 import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
 
 /**
- * SimpleQuery is a query parser that acts similar to a query_string
- * query, but won't throw exceptions for any weird string syntax.
+ * SimpleQuery is a query parser that acts similar to a query_string query, but
+ * won't throw exceptions for any weird string syntax.
+ * 
+ * For more detailed explanation of the query string syntax see also the <a
+ * href=
+ * "https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html"
+ * > online documentation</a>.
  */
-public class SimpleQueryStringBuilder extends QueryBuilder {
+public class SimpleQueryStringBuilder extends QueryBuilder<SimpleQueryStringBuilder> implements BoostableQueryBuilder<SimpleQueryStringBuilder> {
+    /** Name for (de-)serialization. */
     public static final String NAME = "simple_query_string";
-    private Map<String, Float> fields = new HashMap<>();
-    private String analyzer;
-    private Operator operator;
+    /** Query text to parse. */
     private final String queryText;
+    /** Boost to apply to resulting Lucene query. */
+    private float boost = 1.0f;
+    /**
+     * Fields to query against. If left empty will query default field,
+     * currently _ALL. Uses a TreeMap to hold the fields so boolean clauses are
+     * always sorted in same order for generated Lucene query for easier
+     * testing.
+     */
+    private final Map<String, Float> fieldsAndWeights = new TreeMap<>();
+    /** If specified, analyzer to use to parse the query text, using registered default if left empty. */
+    private String analyzer;
+    /** Name of the query. */
     private String queryName;
+    /** Default operator to use for linking boolean clauses. */
+    private Operator defaultOperator;
+    /** If result is a boolean query, minimumShouldMatch parameter to apply. Ignored otherwise. */
     private String minimumShouldMatch;
+    /** Any search flags to be used, ALL by default. */
     private int flags = -1;
-    private Boolean lowercaseExpandedTerms;
-    private Boolean lenient;
-    private Boolean analyzeWildcard;
-    private Locale locale;
-    static final SimpleQueryStringBuilder PROTOTYPE = new SimpleQueryStringBuilder(null);
+    /** Further search settings. */
+    private Settings settings = new Settings();
 
     /**
-     * Operators for the default_operator
+     * This instance is used to de-serialize {@link SimpleQueryStringBuilder}
+     * objects only. During de-serialization a new object will be created that
+     * holds the deserialized query object. As a result seeing the queryText the
+     * prototype is initialized with in any kind of query logs indicates an
+     * implemenation error.
      */
+    static final SimpleQueryStringBuilder PROTOTYPE = new SimpleQueryStringBuilder(
+            "You should never read this query in your logs unless there is an implementation error in ES.");
```

Comment:
I was also going to support passing "null" as I did in most of the other Prototype constants, but now I looked at some registry map containing these builders in a debugger and realized that at least toString() can be called on these prototypes, (calling doXContent()) and this will lead to strange NPEs. Wonder if we can pass in sensible defaults but still make the prototype object unusable otherwise.